src/amo/reducers/home.js

```diff
@@ -293,11 +283,36 @@ export const createInternalSecondaryHeroModule = (
   };
 };
 
-export const createInternalHeroShelves = (
-  heroShelves: ExternalHeroShelvesType,
+export const createInternalShelf = (
+  result: ExternalResultShelfType,
+  lang: string,
+): ResultShelfType => {
+  const shelfAddons = result.addons.map((addon) => {
+    return createInternalAddon(addon, lang);
+  });
+
+  return {
+    title: selectLocalizedContent(result.title, lang),
+    url: result.url,
+    endpoint: result.endpoint,
+    criteria: result.criteria,
+    footer: result.footer
+      ? createInternalHeroCallToAction(result.footer, lang)
+      : null,
+    addons: shelfAddons,
+  };
+};
+
+export const createInternalHomeShelves = (
+  homeShelves: ExternalHomeShelvesType,
   lang: string,
-): HeroShelvesType => {
-  const { primary, secondary } = heroShelves;
+): HomeShelvesType => {
+  const { results, primary, secondary } = homeShelves;
+
+  let customShelves: Array<ResultShelfType> | null = null;
+  if (results !== null) {
+    customShelves = results.map((result) => createInternalShelf(result, lang));
+  }
 
   let secondaryShelf: SecondaryHeroShelfType | null = null;
   if (secondary !== null) {
```

Comment:
Same here, when would `secondary` be `null`?

Reply:
Lines 302-315 were added by Andrew for the update to handle when the API returned localized fields from the API...